lib/services/local/access_list.go

```diff
@@ -303,6 +320,84 @@ func (a *AccessListService) GetAccessListMember(ctx context.Context, accessList
 	return member, trace.Wrap(err)
 }
 
+// GetAccessListOwners returns a list of all owners in an Access List, including those inherited from nested Access Lists.
+//
+// Returned Owners are not validated for ownership requirements – use `IsAccessListOwner` for validation.
+func (a *AccessListService) GetAccessListOwners(ctx context.Context, accessListName string) ([]*accesslist.Owner, error) {
+	seen := map[string]struct{}{}
+	type queueItem struct {
+		name  string
+		depth int
+	}
+	queue := []queueItem{{name: accessListName, depth: 0}}
+	var allOwners []*accesslist.Owner
+
+	for len(queue) > 0 {
+		item := queue[0]
+		queue = queue[1:]
+
+		if item.depth > accesslist.MaxAllowedDepth {
+			a.log.Warnf("Exceeded maximum depth of %d while gathering nested owners for access list %s", accesslist.MaxAllowedDepth, accessListName)
+			continue
+		}
+
+		accessList, err := a.GetAccessList(ctx, item.name)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		// Owners are inherited from members of nested lists.
+		// So, on l0, we want to grab all the owners from the provided accessListName,
+		// while > l0, we want to fetch members of the nested lists.
+		if item.name == accessListName {
+			for _, owner := range accessList.GetOwners() {
+				if owner.MembershipKind == accesslist.MembershipKindList {
+					if _, ok := seen[owner.Name]; ok {
+						continue
+					}
+					seen[owner.Name] = struct{}{}
+					queue = append(queue, queueItem{name: owner.Name, depth: item.depth + 1})
+				} else {
+					allOwners = append(allOwners, &owner)
+				}
+			}
+		} else {
```

Comment:
can we simplify the loop and deflaten the `} else {
```suggestion
              continue
		}
```

so instead of 
```go
for _, v := elemst {
   if ... {
   } else {
     logic
   }
}

return all
```

we will have: 

```go
for _, v := elemst {
   if ... {
     continue
   }
   // logic
}

return all
```